File: Módulo 2 - Array/Readability/readability.c
// =================== //
//                     //
//                     //
// === NOT WORKING === //
//                     //
//                     //
// =================== //

#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Submit a Text: ");

    int letters = 0;
    int words = 1;
    int sentences = 0;

    //Find text infos
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if(isalpha(text[i]))
        {
            letters++;
        }
        else if(isspace(text[i]))
        {
            words++;
        }
        else if(text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    //Execution of the formula
    float l = (float)letters / (float)words * 100;
    float s = (float)sentences / (float)words * 100;

    float result = round(0.0588 * l - 0.296 * s - 15.8);

    if(result < 1)
    {
        printf("Before Grade 1\n");
    }
    if(result >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) result);
    }
}

File: Module 6 - Python/readability/readability.py
from cs50 import get_string

def count_letters(text):
    cnt = 0
    for i in range(len(text)):
        if text[i].isalnum():
            cnt += 1;
    return cnt

def count_words(text):
    cnt = 0
    for i in range(len(text)):
        if text[i].isspace():
            cnt += 1
    return cnt + 1

def count_sentences(text):
    cnt = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            cnt += 1
    return cnt

text = get_string("Text: ");

l = count_letters(text) * 100.0 / count_words(text)
s = count_sentences(text) * 100.0 / count_words(text)
grade = int(round(0.0588 * l - 0.296 * s - 15.8))

if grade < 1:
    print(f"Before Grade 1\n")
elif grade > 16:
    print(f"Grade 16+\n")
else:
    print(f"Grade {grade}\n")